File: DimoTalk.Maui/Models/Message.cs
namespace DimoTalk.Maui.Models;

public enum MessageRole { System, User, Assistant }

public class Message
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public MessageRole Role { get; set; }
    public string Content { get; set; } = string.Empty;
    public DateTime Timestamp { get; set; } = DateTime.Now;

    public string RoleName => Role.ToString().ToLowerInvariant();

    public override string ToString() => $"{Role}: {Content}";
}

File: Hardwaver/lib/models/message.dart
enum MessageRole { system, user, assistant }

class Message {
  final String id;
  final MessageRole role;
  final String content;
  final DateTime timestamp;

  const Message({
    required this.id,
    required this.role,
    required this.content,
    required this.timestamp,
  });

  Message copyWith({
    String? id,
    MessageRole? role,
    String? content,
    DateTime? timestamp,
  }) {
    return Message(
      id: id ?? this.id,
      role: role ?? this.role,
      content: content ?? this.content,
      timestamp: timestamp ?? this.timestamp,
    );
  }

  Map<String, dynamic> toOpenAIMap() => {
        'role': role.name,
        'content': content,
      };
}
